mybackend/controller/competition.js

```diff
@@ -0,0 +1,283 @@
+                {},
+                {
+                    headers: {
+                        Authorization: `token ${githubToken}`,
+                        Accept: 'application/vnd.github.v3+json'
+                    }
+                }
+            );
+
+            res.status(200).json(competition);
+
+        } catch (error) {
+            console.error('Error assigning judge role:', error.message);
+            res.status(500).json({ message: error.message });
+        }
+    }
+
+    approveAndMergePR = async (req, res) => {
+        const { id } = req.params;
+        const { prNumber } = req.body;
+
+        try {
+            const competition = await Competition.findById(id);
+            if (!competition) {
+                return res.status(404).json({ message: 'Project not found' });
+            }
+
+            const repoUrl = new URL(competition.repositoryLink).pathname.substring(1);
+            const prResponse = await axios.get(`https://api.github.com/repos/${repoUrl}/pulls/${prNumber}`, {
+                headers: {
+                    Authorization: `token ${process.env.GITHUB_ADMIN_TOKEN}`,
+                },
+            });
+
+            const { labels, user: { login } } = prResponse.data;
+            const user = await User.findOne({ username: login });
+
+            if (!user) {
+                return res.status(404).json({ message: 'User not found' });
+            }
+
+            let reward = 0;
+            let updateData = {};
+
+            if (labels.some(label => label.name.toLowerCase() === 'feature')) {
+                reward = competition.rewards.feature;
+                updateData = { $inc: { totalEarnings: reward, Features: 1 } };
+            } else if (labels.some(label => label.name.toLowerCase() === 'bug')) {
+                reward = competition.rewards.bug;
+                updateData = { $inc: { totalEarnings: reward, Bugs: 1 } };
+            } else if (labels.some(label => label.name.toLowerCase() === 'optimization')) {
+                reward = competition.rewards.optimization;
+                updateData = { $inc: { totalEarnings: reward, Optimisations: 1 } };
+            } else if (labels.some(label => label.name.toLowerCase() === 'security')) {
+                reward = competition.rewards.security;
+                updateData = { $inc: { totalEarnings: reward } };
+            }
+
+            await axios.put(
+                `https://api.github.com/repos/${repoUrl}/pulls/${prNumber}/merge`,
+                {},
+                {
+                    headers: {
+                        Authorization: `token ${process.env.GITHUB_ADMIN_TOKEN}`,
+                    },
+                }
+            );
+
+            await User.findByIdAndUpdate(user._id, updateData);
+
+            res.status(200).json({ message: 'PR merged and user earnings updated' });
+        } catch (error) {
+            console.error('Error merging PR:', error.message);
```

Comment:
`console.error()` statements are useful for development, but consider providing user-friendly error messages instead of console logs. A `setError` state that shows relevant messages in the UI would improve error visibility and handling. Because most users won't open the dev console to debug for errors.